src/mango/src/mango_idx_view.erl

```diff
@@ -417,7 +417,7 @@ range(_, _, LCmp, Low, HCmp, High) ->
 % beginsWith requires both a high and low bound
 range({[{<<"$beginsWith">>, Arg}]}, LCmp, Low, HCmp, High) ->
     {LCmp0, Low0, HCmp0, High0} = range({[{<<"$gte">>, Arg}]}, LCmp, Low, HCmp, High),
-    range({[{<<"$lte">>, <<Arg/binary, 16#10FFFF>>}]}, LCmp0, Low0, HCmp0, High0);
+    range({[{<<"$lte">>, <<Arg/binary, 16#10FFFF/utf8>>}]}, LCmp0, Low0, HCmp0, High0);
```

Comment:
Would `16#FFFF/utf8` work here?

Reply:
@nickva that's what I used in the first iteration - is there a preference for `16#FFFF/utf8` over `16#10FFFF/utf8`?

Reply:
I changed it in response to https://github.com/apache/couchdb/pull/4810#issuecomment-1790609445

Reply:
@nickva see [my comment](https://github.com/apache/couchdb/pull/4810#issuecomment-1790609445) on #4810:

_"Erlang itself supports `U+10FFFF` and it is numerically greater than `U+FFFF`"_ --  while I accept that `U+FFFF` might be the Unicode-official marker, Erlang's support for a higher range worries me a bit.

Reply:
I think `U+FFFF` is the canonical "highest sorting" string according to the collator rules, `U+10FFFF` is just the highest unicode code point numerically but not sure if is guaranteed to be the highest sorting one. Collators can be tailed in various ways and such.

Verifying that `U+FFFF` sorts higher than `U+10FFFF`
```erlang
(node1@127.0.0.1)80> mango_json:cmp(<<16#10FFFF/utf8>>, <<16#FFFF/utf8>>).
-1
```

Reply:
Okay, thanks for the confirmation.  Then `U+FFFF` is the constant to go.

Reply:
For the sake of diligence, I have launched another the full CI run with `U+FFFF` in #4829.

Reply:
thanks @nickva @pgj. I pushed an update to use `U+FFFF` and fixed the unit test for invalid UTF8 strings.

Reply:
> "Erlang itself supports U+10FFFF and it is numerically greater than U+FFFF" -- while I accept that U+FFFF might be the Unicode-official marker, Erlang's support for a higher range worries me a bit.

Perhaps confusingly unicode collation rules don't sort by code point values. Higher code-points may sort lower. Besides just `U+10FFFF` and `U+FFFF` there are other such example:

```erlang
mango_json:cmp(<<16#1D175/utf8>>, <<16#41/utf8>>).
-1
```

Now, there may not be any other characters higher than U+10FFFF and lower than U+FFFF, so either one is probably ok in practice, but it's a bit neater if we can use the canonical max sorting character. Jiffy / erlang / couch_util:validate_utf8/1 should pass through both of those characters, so technically users could insert keys with those crazy values in them:

```erlang
<<16#FFFF/utf8>> = jiffy:decode(<<"\"\\uFFFF\"">>).
```

Reply:
Yes, that completely makes sense.  In retrospect, it is quite logical that the collation algorithm may define arbitrary ordering over the values independently of their numerical representation.  It may not be that efficient though.

I am somewhat old school in terms of character encodings (despite being a person for who the letters in the native language does not fit in 7-bit ASCII – ISO-8859-2 for the win! But I was able to do the move to IPv6 recently so there is still hope for me... :smile:)

Reply:
All that's missing now is for you to grow a hipster beard 😜

Reply:
While my brain has proven itself resourceful enough in the recent decades (I am an arts guy who became an engineer because of the love creating things) my facial hair has not.

Reply:
> All that's missing now is for you to grow a hipster beard 😜

There is a unicode code point for it! &#129492; (`U+1F9D4`)

Reply:
🤣 that should go into one test 😉

Reply:
Thanks @nickva !